src/smart-components/group/add-group/set-name.js

```diff
@@ -0,0 +1,81 @@
+import React, { Fragment, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Stack, StackItem } from '@patternfly/react-core';
+import useFieldApi from '@data-driven-forms/react-form-renderer/dist/esm/use-field-api';
+import useFormApi from '@data-driven-forms/react-form-renderer/dist/esm/use-form-api';
+import { TextInput } from '@patternfly/react-core/dist/esm/components/TextInput/TextInput';
+import { TextArea } from '@patternfly/react-core/dist/esm/components/TextArea/TextArea';
+import { FormGroup } from '@patternfly/react-core/dist/esm/components/Form/FormGroup';
+import { debouncedAsyncValidator } from '../validators';
+
+const SetName = (props) => {
+  const { input } = useFieldApi(props);
+  const formOptions = useFormApi();
+  const { 'group-name': name, 'group-description': description } = formOptions.getState().values;
+  const [groupName, setGroupName] = useState(name || '');
+  const [groupNameError, setGroupNameError] = useState();
+  const [groupDescription, setGroupDescription] = useState(description);
+
+  const groupNameValidated = () => (groupName === undefined || groupNameError ? 'error' : 'default');
+  const groupDescriptionValidated = () => (groupDescription && groupDescription.length > 150 ? 'error' : 'default');
+
+  const processGroupName = (value) => {
+    input.onChange(undefined);
+    debouncedAsyncValidator(value)
+      .then(() => {
+        input.onChange(value);
+        setGroupNameError(undefined);
+      })
+      .catch((error) => {
+        setGroupNameError(error);
+      });
+    setGroupName(value);
+  };
+
+  useEffect(() => {
+    groupName.length > 0 && processGroupName(groupName);
+  }, []);
+
+  return (
+    <Fragment>
+      <Stack hasGutter>
+        <StackItem className="ins-c-rbac__summary">
+          <FormGroup label="Group name" helperTextInvalid={groupName ? groupNameError : 'Required'} isRequired validated={groupNameValidated()}>
+            <TextInput
+              value={groupName}
+              type="text"
+              validated={groupNameValidated()}
+              onBlur={() => {
+                groupName === '' && setGroupName(undefined);
+              }}
+              onChange={(value) => processGroupName(value)}
+              aria-label="Group name"
+            />
+          </FormGroup>
+        </StackItem>
+        <StackItem>
+          <FormGroup label="Group description" helperTextInvalid="Can have maximum of 150 characters." validated={groupDescriptionValidated()}>
```

Comment:
@mmenestr is the text correct? `Can have maximum of 150 characters.` I think it would be better to use `Group description can have maximum of 150 characters`